### PR TITLE
Jiayu Sort projects by most recent edited inventories and members

### PR DESCRIPTION
**Pull request by peter6866**
# Description
image

This is only for sorting the members since inventory is not in use.

## Related PRS (if any):
To test this PR you need to checkout [backend PR1025](https://github.com/OneCommunityGlobal/HGNRest/pull/1025).

## Main changes explained:
- Update `ProjectTableHeader.jsx` to include sorting button
- Update `Projects.jsx` to include correct comparison logic for sorting

## How to test:
1. check into current branch
2. do `npm install` and `npm run start:local` to run this PR locally
3. Clear site data/cache
4. log as owner/admin user
5. go to dashboard→ Other Links -> Projects
6. select a project except the first one and add a member
7. go back to projects page and click on the sort button next to the members header
8. verify that the project you've added the member to is now the first project
9. click again the button to reset the sorting
10. add a member to another project and verify that it goes to the top after sorting
11. go to the project you added the member first, click into the user profile of the member and in their projects tab remove the member from the project. 
12. go back to projects page and click the sort button.
13. verify that this project goes to the top again.
14. **You can also follow the video bellow for testing steps.**

## Screenshots or videos of changes:
https://www.loom.com/share/f90a2754cee444f8ae711a0a0f09c749?sid=3a343985-0bc6-480a-9942-b18bda25ec1e

## Note:
Include the information the reviewers need to know.

### CODE REVIEW

**Review by aaryaneil — DISMISSED** [review]
Tested the PR and after sorting the projects list based on the members it doesn't sort based on the project to which the person is added to latest

**Comment by peter6866** [comment]
> Tested the PR and after sorting the projects list based on the members it doesn't sort based on the project to which the person is added to latest

What have you tested? Can you provide some screenshots?

**Review by JatinAgrawal94 — DISMISSED** [review]
Please fix this UI issue.
Screenshot 2024-07-16 at 3 13 13 PM

**Comment by peter6866** [comment]
> Please fix this UI issue. Screenshot 2024-07-16 at 3 13 13 PM

This is not related to this PR. Please check the development branch first then test.

**Review by nikhilpittala16 — APPROVED** [review]
I have tested the PR and everything works as expected.
Screenshot (123)
Screenshot (125)
Screenshot (127)
Screenshot (126)
Screenshot (128)
Screenshot (129)
Screenshot (130)

**Review by vishavk1992 — APPROVED** [review]
I have tested this PR with Backend PR1025 . it works as expected and i was able to aSort projects by most recent edited inventories and members.


https://github.com/user-attachments/assets/e68bcd75-5288-45ea-94f3-f80e5ec8066c

**Review by Chetan0606 — APPROVED** [review]
I have tested the PR. I have used my owner account to login and test. It is working as expected.
Screenshot (298)
Screenshot (297)

**Comment by kabraambika** [comment]
**Proposed Improvement:**
Sort order should be preserved on projects when no changes are made on the members page.

**Please refer to the video below for a detailed explanation of this improvement.**

https://github.com/user-attachments/assets/720435e2-af06-4d82-916f-d61e9e346a36

**Steps to Recreate the Scenario:**
- The top project is "Ambika-2443-PR-test" with the member Ambika Kabra, and the second project is "NewProject Ambika" with members Ambika Kabra, Niketha Anand, and Niketha Anand.
- Click on "Sort" for members, and "NewProject Ambika" will appear on top because its members were recently updated.
- Click on the "Members" button to check the members of "NewProject Ambika" again.
- Observe no changes on the members page and return to the Projects page.
- Notice that "Ambika-2443-PR-test" is back on top.

**Suggested Behavior:**
The sort order should remain enabled if no changes are made to a project's members, ensuring that "NewProject Ambika" remains visible on top.

**Review by SrikanthChowdary495 — COMMENTED** [review]
I have tested this PR with Backend PR1025. Everything works as expected. I was able to sort the projects after adding the members  and removing the members from the project.

File: src/components/Projects/ProjectTableHeader/ProjectTableHeader.jsx [hunk 1]
```diff
@@ -65,10 +65,23 @@ const ProjectTableHeader = props => {
        </span> 
       </th>
       <th scope="col" id="projects__inv">
-        {INVENTORY}
+        <span className='d-flex justify-content-between'>
+          {INVENTORY}
+          {/* <Button size='sm' className='ml-2' id='SortingByRecentEditedInventory' onClick={props.handleSort}>
```
Comment (kabraambika) [inline src/components/Projects/ProjectTableHeader/ProjectTableHeader.jsx:70]: This code will never be executed and can be removed.

**Review by kabraambika — CHANGES_REQUESTED** [review]

File: src/components/Projects/ProjectTableHeader/ProjectTableHeader.jsx (hunk 1 above)
Comment (peter6866) [reply src/components/Projects/ProjectTableHeader/ProjectTableHeader.jsx:70]: This is a needed feature in the future and should be kept.

**Review by kabraambika — APPROVED** [review]
I have reviewed the suggested changes in the frontend and completed the review for the associated backend PR as well. Aside from the suggested improvement, my previous comment have been addressed. Therefore, I am approving the PR. Thanks!

**Review by SrikanthChowdary495 — APPROVED** [review]

**Comment by beblicarl** [comment]
The functionality works as intended and the code is great

[screencast-bpconcjcammlapcogcnnelfmaeghhagj-2024.07.19-10_04_33.webm](https://github.com/user-attachments/assets/89ad61d8-5169-45aa-836a-fb048e9823ac)

**Review by beblicarl — APPROVED** [review]

**Review by ishan-miglani — APPROVED** [review]
I have verified the changes in this PR. the functionality works correctly. I am able to sort projects based on the most recently edited members. Great work!

https://github.com/user-attachments/assets/f1532691-3bc2-4bf3-b8eb-8ee588ed5560

**Comment by souravwalke** [comment]
I have tested this PR with Backend PR1025. Everything works as expected. I was able to sort the projects perfectly. 
Screenshot 2024-07-19 at 4 41 41 PM
Screenshot 2024-07-19 at 4 42 20 PM
Screenshot 2024-07-19 at 4 42 41 PM

**Review by jinxyou — APPROVED** [review]
I tested the adding members and sorting functions. All the functions are working fine. GReat work! 
Screenshot 2024-07-20 190806
Screenshot 2024-07-20 190816
Screenshot 2024-07-20 190825

**Review by cvtqx — APPROVED** [review]
Functionality works as described. When making changes (adding or deleting members from a team) the team moves to the top of the list on sort.

https://github.com/user-attachments/assets/2b0ebb51-ba1f-40d8-8e15-8d79b413eb8f

**Comment by one-community** [comment]
Thank you all, merging!

**Pull request merged**